File: 2022-23/contest1/prog/intermediate/next_base.java
import java.io.*;
import java.util.*;
import java.lang.Math;

public class next_base {

	public static long str_to_int(String str, long base) {
		long res = 0;
		for(int i = 0, N =  str.length(); i < N; ++i) {
			res += ((str.charAt(N-i-1) - '0') * (long)(Math.pow(base, i)));
		}
		return res;
	}

	public static void main(String args[]) {
		Scanner input = new Scanner(System.in);
		for(long k = 0; k < 5; ++k) {
			long n = input.nextLong();
			int b = input.nextInt();
			String s = input.next();
			input.nextLine();
			long num = str_to_int(s, b);
			int[] cnts = new int[10];
			for(long i = 0; i < n; ++i) {
				long numi = num + i;
				while(numi > 0) {
					int d = (int)(numi % b);
					++cnts[d];
					numi /= b;
				}
			}
			System.out.println(cnts[b-1]);
		}
	}
}

File: 2022-23/contest1/prog/intermediate/next_base.py
import os
import sys

def str_to_int(s, base):
	res = 0
	N = len(s)
	for i in range(N):
		res += (int(s[N-i-1]) * pow(base, i))
	return res

for k in range(5):
	args = [x for x in input().split()]
	n = int(args[0])
	b = int(args[1])
	s = args[2]
	num = str_to_int(s, b)
	tot = 0
	cnts = [0]*b
	for i in range(n):
		numi = num + i
		while numi > 0:
			d = numi % b
			++cnts[d]
			numi = numi // b
	print(cnts[b-1])
